### ISSUE TITLE
Kernel Panic on AWS (disk related)

### ISSUE THREAD

**Issue by peter-styk-ext** [jangaraj/zenoss5-core-autodeploy#8]
Love the script. Hopefully this will help someone else. Running on Centos7 eu-west-1 ami-1c2e8b6b in VPC on public Subnet as m3.2xlarge (30GB). Using 5 separate devices attached in AWS console. Each 50GB (block devices, not on instance so images can be saved). Starting with:
```
sudo -s
yum update -y
```
then downloading the script and starting installation.
```
./core-autodeploy.sh -d /dev/xvdb -s /dev/xvdc -v /dev/xvdd -b /dev/xvde
```
...then accepting destroy of data on those volumes 4 times...script exits once during installation of NTP with access denied. on repeat of above command it finishes successfully.
```
5 Final overview - (Fri, 12 Jun 2015 11:10:54 +0000)
Control Center & Zenoss Core 5 installation completed
Set password for Control Center ccuser user: passwd ccuser
Please visit Control Center https://54.154.105.149/ in your favorite web browser to complete setup, log in with ccuser user
Add following line to your hosts file:
54.154.105.149 ip-10-20-0-143.eu-west-1.compute.internal hbase.ip-10-20-0-143.eu-west-1.compute.internal opentsdb.ip-10-20-0-143.eu-west-1.compute.internal rabbitmq.ip-10-20-0-143.eu-west-1.compute.internal zenoss5.ip-10-20-0-143.eu-west-1.compute.internal
or edit /etc/default/serviced and set SERVICED_VHOST_ALIASES to your FQDN
Please also reboot machine, because SELINUX is still active!
```
About 5 minutes after installation completes, system freezes and cannot be accessed on its own just by being idle. At this point I have installed brand new clean install 3 times. 3rd time I tried it i was quick enough to confirm https address and saw the login screen. Quick list of fstab revealed...
```
[ec2-user@ip-10-20-0-143 ~]$ cat /etc/fstab

#
# /etc/fstab
# Created by anaconda on Tue Jul  8 16:48:19 2014
#
# Accessible filesystems, by reference, are maintained under '/dev/disk'
# See man pages fstab(5), findfs(8), mount(8) and/or blkid(8) for more info
#
UUID=d40838c5-c088-41fa-a41e-82681315ba79 /                       xfs     defaults        1 1
/dev/xvdb /var/lib/docker btrfs rw,noatime,nodatacow 0 0
/dev/xvdc /opt/serviced/var xfs defaults,noatime 0 0
/dev/xvdd /opt/serviced/var/volumes btrfs rw,noatime,nodatacow 0 0
/dev/xvde /opt/serviced/var/backups xfs defaults,noatime 0 0
```
Since no access to instance is possible, reboot and stop/restart produces same result (inaccessible instance) i pulled AWS startup log and discover Kernel Panic (due to volumes not mounting properly or missing drivers). Perhaps you've seen this before, perhaps I should be using same disk and different partitions instead? Please let me know if this rings any bells. 

Below is output of instance startup
```
[    0.000000] Initializing cgroup subsys cpuset
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] Linux version 3.10.0-229.4.2.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.2 20140120 (Red Hat 4.8.2-16) (GCC) ) #1 SMP Wed May 13 10:06:09 UTC 2015
[    0.000000] Command line: ro root=UUID=d40838c5-c088-41fa-a41e-82681315ba79 console=hvc0 LANG=en_US.UTF-8
[    0.000000] ACPI in unprivileged domain disabled
[    0.000000] e820: BIOS-provided physical RAM map:
[    0.000000] Xen: [mem 0x0000000000000000-0x000000000009ffff] usable
[    0.000000] Xen: [mem 0x00000000000a0000-0x00000000000fffff] reserved
[    0.000000] Xen: [mem 0x0000000000100000-0x00000007807fffff] usable
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] DMI not present or invalid.
[    0.000000] No AGP bridge found
[    0.000000] e820: last_pfn = 0x780800 max_arch_pfn = 0x400000000
[    0.000000] e820: last_pfn = 0x100000 max_arch_pfn = 0x400000000
[    0.000000] init_memory_mapping: [mem 0x00000000-0x000fffff]
[    0.000000] init_memory_mapping: [mem 0x77fe00000-0x77fffffff]
[    0.000000] init_memory_mapping: [mem 0x77c000000-0x77fdfffff]
[    0.000000] init_memory_mapping: [mem 0x700000000-0x77bffffff]
[    0.000000] init_memory_mapping: [mem 0x00100000-0x6ffffffff]
[    0.000000] init_memory_mapping: [mem 0x780000000-0x7807fffff]
[    0.000000] NUMA turned off
[    0.000000] Faking a node at [mem 0x0000000000000000-0x00000007807fffff]
[    0.000000] Initmem setup node 0 [mem 0x00000000-0x7807fffff]
[    0.000000]   NODE_DATA [mem 0x77c3ed000-0x77c413fff]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x00001000-0x00ffffff]
[    0.000000]   DMA32    [mem 0x01000000-0xffffffff]
[    0.000000]   Normal   [mem 0x100000000-0x7807fffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x00001000-0x0009ffff]
[    0.000000]   node   0: [mem 0x00100000-0x7807fffff]
[    0.000000] SFI: Simple Firmware Interface v0.81 http://simplefirmware.org
[    0.000000] smpboot: Allowing 8 CPUs, 0 hotplug CPUs
[    0.000000] No local APIC present
[    0.000000] APIC: disable apic facility
[    0.000000] APIC: switched to apic NOOP
[    0.000000] PM: Registered nosave memory: [mem 0x000a0000-0x000fffff]
[    0.000000] e820: cannot find a gap in the 32bit address range
[    0.000000] e820: PCI devices with unassigned 32bit BARs may break!
[    0.000000] e820: [mem 0x780900000-0x780cfffff] available for PCI devices
[    0.000000] Booting paravirtualized kernel on Xen
[    0.000000] Xen version: 4.2.amazon (preserve-AD)
[    0.000000] setup_percpu: NR_CPUS:5120 nr_cpumask_bits:8 nr_cpu_ids:8 nr_node_ids:1
[    0.000000] PERCPU: Embedded 28 pages/cpu @ffff88077c000000 s82752 r8192 d23744 u262144
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 7743338
[    0.000000] Policy zone: Normal
[    0.000000] Kernel command line: ro root=UUID=d40838c5-c088-41fa-a41e-82681315ba79 console=hvc0 LANG=en_US.UTF-8
[    0.000000] PID hash table entries: 4096 (order: 3, 32768 bytes)
[    0.000000] Checking aperture...
[    0.000000] No AGP bridge found
[    0.000000] Memory: 30822292k/31465472k available (6245k kernel code, 388k absent, 642792k reserved, 4177k data, 1604k init)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=8, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] 	RCU restricting CPUs from NR_CPUS=5120 to nr_cpu_ids=8.
[    0.000000] 	Experimental no-CBs for all CPUs
[    0.000000] 	Experimental no-CBs CPUs: 0-7.
[    0.000000] NR_IRQS:327936 nr_irqs:336 16
[    0.000000] Console: colour dummy device 80x25
[    0.000000] console [tty0] enabled
[    0.000000] console [hvc0] enabled
[    0.000000] allocated 126353408 bytes of page_cgroup
[    0.000000] please try 'cgroup_disable=memory' option if you don't want memory cgroups
[    0.000000] installing Xen timer for CPU 0
[    0.000000] tsc: Detected 2500.050 MHz processor
[    0.001000] Calibrating delay loop (skipped), value calculated using timer frequency.. 5000.10 BogoMIPS (lpj=2500050)
[    0.001000] pid_max: default: 32768 minimum: 301
[    0.001000] Security Framework initialized
[    0.001000] SELinux:  Initializing.
[    0.004000] Dentry cache hash table entries: 4194304 (order: 13, 33554432 bytes)
[    0.011580] Inode-cache hash table entries: 2097152 (order: 12, 16777216 bytes)
[    0.014396] Mount-cache hash table entries: 4096
[    0.014640] Initializing cgroup subsys memory
[    0.014655] Initializing cgroup subsys devices
[    0.014660] Initializing cgroup subsys freezer
[    0.014663] Initializing cgroup subsys net_cls
[    0.014667] Initializing cgroup subsys blkio
[    0.014670] Initializing cgroup subsys perf_event
[    0.014674] Initializing cgroup subsys hugetlb
[    0.014730] ENERGY_PERF_BIAS: Set to 'normal', was 'performance'
[    0.014730] ENERGY_PERF_BIAS: View and update with x86_energy_perf_policy(8)
[    0.014738] CPU: Physical Processor ID: 0
[    0.014740] CPU: Processor Core ID: 0
[    0.015437] Last level iTLB entries: 4KB 512, 2MB 0, 4MB 0
[    0.015437] Last level dTLB entries: 4KB 512, 2MB 0, 4MB 0
[    0.015437] tlb_flushall_shift: 6
[    0.043171] ftrace: allocating 23915 entries in 94 pages
[    0.048069] cpu 0 spinlock event irq 17
[    0.048124] Performance Events: unsupported p6 CPU model 62 no PMU driver, software events only.
[    0.049283] NMI watchdog: disabled (cpu0): hardware events not enabled
[    0.049346] installing Xen timer for CPU 1
[    0.049362] cpu 1 spinlock event irq 24
[    0.049415] SMP alternatives: switching to SMP code
[    0.074569] installing Xen timer for CPU 2
[    0.074580] cpu 2 spinlock event irq 31
[    0.075430] installing Xen timer for CPU 3
[    0.075441] cpu 3 spinlock event irq 38
[    0.076280] installing Xen timer for CPU 4
[    0.076292] cpu 4 spinlock event irq 45
[    0.077207] installing Xen timer for CPU 5
[    0.077221] cpu 5 spinlock event irq 52
[    0.078082] installing Xen timer for CPU 6
[    0.078093] cpu 6 spinlock event irq 59
[    0.078994] installing Xen timer for CPU 7
[    0.079000] cpu 7 spinlock event irq 66
[    0.079800] Brought up 8 CPUs
[    0.080003] devtmpfs: initialized
[    0.083476] EVM: security.selinux
[    0.083479] EVM: security.ima
[    0.083481] EVM: security.capability
[    0.084191] atomic64 test passed for x86-64 platform with CX8 and with SSE
[    0.084245] xen:grant_table: Grant tables using version 2 layout
[    0.084259] Grant table initialized
[    0.084349] NET: Registered protocol family 16
[    0.084423] PCI: setting up Xen PCI frontend stub
[    0.086080] ACPI: Interpreter disabled.
[    0.086080] xen:balloon: Initialising balloon driver
[    0.087015] xen_balloon: Initialising balloon driver
[    0.087030] vgaarb: loaded
[    0.087076] SCSI subsystem initialized
[    0.087107] usbcore: registered new interface driver usbfs
[    0.087118] usbcore: registered new interface driver hub
[    0.087125] usbcore: registered new device driver usb
[    0.087125] PCI: System does not support PCI
[    0.087125] PCI: System does not support PCI
[    0.087125] NetLabel: Initializing
[    0.087125] NetLabel:  domain hash size = 128
[    0.087126] NetLabel:  protocols = UNLABELED CIPSOv4
[    0.087139] NetLabel:  unlabeled traffic allowed by default
[    0.087158] Switching to clocksource xen
[    0.091278] pnp: PnP ACPI: disabled
[    0.093289] NET: Registered protocol family 2
[    0.093661] TCP established hash table entries: 262144 (order: 9, 2097152 bytes)
[    0.094092] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes)
[    0.094221] TCP: Hash tables configured (established 262144 bind 65536)
[    0.165713] TCP: reno registered
[    0.165781] UDP hash table entries: 16384 (order: 7, 524288 bytes)
[    0.165927] UDP-Lite hash table entries: 16384 (order: 7, 524288 bytes)
[    0.166066] NET: Registered protocol family 1
[    0.166167] platform rtc_cmos: registered platform RTC device (no PNP device found)
[    0.166369] RAPL PMU detected, hw unit 2^-16 Joules, API unit is 2^-32 Joules, 3 fixed counters 163840 ms ovfl timer
[    0.166876] microcode: CPU0 sig=0x306e4, pf=0x1, revision=0x415
[    0.166886] microcode: CPU1 sig=0x306e4, pf=0x1, revision=0x415
[    0.166898] microcode: CPU2 sig=0x306e4, pf=0x1, revision=0x415
[    0.166914] microcode: CPU3 sig=0x306e4, pf=0x1, revision=0x415
[    0.166927] microcode: CPU4 sig=0x306e4, pf=0x1, revision=0x415
[    0.166941] microcode: CPU5 sig=0x306e4, pf=0x1, revision=0x415
[    0.166956] microcode: CPU6 sig=0x306e4, pf=0x1, revision=0x415
[    0.166970] microcode: CPU7 sig=0x306e4, pf=0x1, revision=0x415
[    0.167005] microcode: Microcode Update Driver: v2.00 <tigran@aivazian.fsnet.co.uk>, Peter Oruba
[    0.167208] futex hash table entries: 2048 (order: 5, 131072 bytes)
[    0.167237] Initialise system trusted keyring
[    0.167285] audit: initializing netlink socket (disabled)
[    0.167294] type=2000 audit(1434112249.317:1): initialized
[    0.185592] HugeTLB registered 2 MB page size, pre-allocated 0 pages
[    0.186364] zbud: loaded
[    0.186515] VFS: Disk quotas dquot_6.5.2
[    0.186544] Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.186685] msgmni has been set to 32768
[    0.186734] Key type big_key registered
[    0.187726] alg: No test for stdrng (krng)
[    0.187741] NET: Registered protocol family 38
[    0.187751] Key type asymmetric registered
[    0.187755] Asymmetric key parser 'x509' registered
[    0.187783] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.187841] io scheduler noop registered
[    0.187846] io scheduler deadline registered (default)
[    0.187867] io scheduler cfq registered
[    0.187934] pci_hotplug: PCI Hot Plug PCI Core version: 0.5
[    0.187947] pciehp: PCI Express Hot Plug Controller Driver version: 0.4
[    0.188356] Serial: 8250/16550 driver, 4 ports, IRQ sharing enabled
[    0.188729] Non-volatile memory driver v1.3
[    0.188735] Linux agpgart interface v0.103
[    0.188796] crash memory driver: version 1.1
[    0.188836] rdac: device handler registered
[    0.188882] hp_sw: device handler registered
[    0.188889] emc: device handler registered
[    0.188894] alua: device handler registered
[    0.188921] libphy: Fixed MDIO Bus: probed
[    0.188966] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    0.188974] ehci-pci: EHCI PCI platform driver
[    0.188983] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    0.188988] ohci-pci: OHCI PCI platform driver
[    0.188999] uhci_hcd: USB Universal Host Controller Interface driver
[    0.189044] usbcore: registered new interface driver usbserial
[    0.189052] usbcore: registered new interface driver usbserial_generic
[    0.189060] usbserial: USB Serial support registered for generic
[    0.189076] i8042: PNP: No PS/2 controller found. Probing ports directly.
[    1.198129] i8042: No controller found
[    1.198221] mousedev: PS/2 mouse device common for all mice
[    1.258399] rtc_cmos rtc_cmos: rtc core: registered rtc_cmos as rtc0
[    1.258445] rtc_cmos: probe of rtc_cmos failed with error -38
[    1.258504] hidraw: raw HID events driver (C) Jiri Kosina
[    1.258582] usbcore: registered new interface driver usbhid
[    1.258585] usbhid: USB HID core driver
[    1.258631] drop_monitor: Initializing network drop monitor service
[    1.258707] TCP: cubic registered
[    1.258712] Initializing XFRM netlink socket
[    1.258781] NET: Registered protocol family 10
[    1.258923] NET: Registered protocol family 17
[    1.258935] mce: Unable to init device /dev/mcelog (rc: -5)
[    1.259091] Loading compiled-in X.509 certificates
[    1.259115] Loaded X.509 cert 'CentOS Linux kpatch signing key: ea0413152cde1d98ebdca3fe6f0230904c9ef717'
[    1.259129] Loaded X.509 cert 'CentOS Linux Driver update signing key: 7f421ee0ab69461574bb358861dbe77762a4201b'
[    1.259571] Loaded X.509 cert 'CentOS Linux kernel signing key: 59445632185e5a18d98cc70f8093293509770427'
[    1.259590] registered taskstats version 1
[    1.259993] Key type trusted registered
[    1.260276] Key type encrypted registered
[    1.260515] IMA: No TPM chip found, activating TPM-bypass!
[    1.260569] xenbus_probe_frontend: Device with no driver: device/vbd/2048
[    1.260573] xenbus_probe_frontend: Device with no driver: device/vbd/2112
[    1.260576] xenbus_probe_frontend: Device with no driver: device/vbd/2096
[    1.260579] xenbus_probe_frontend: Device with no driver: device/vbd/2080
[    1.260582] xenbus_probe_frontend: Device with no driver: device/vbd/2064
[    1.260585] xenbus_probe_frontend: Device with no driver: device/vif/0
[    1.260644] drivers/rtc/hctosys.c: unable to open rtc device (rtc0)
[    1.260727] md: Waiting for all devices to be available before autodetect
[    1.260732] md: If you don't use raid, use raid=noautodetect
[    1.260865] md: Autodetecting RAID arrays.
[    1.260870] md: Scanned 0 and added 0 devices.
[    1.260872] md: autorun ...
[    1.260874] md: ... autorun DONE.
[    1.260898] List of all partitions:
[    1.260901] No filesystem could mount root, tried: 
[    1.260905] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.260909] CPU: 1 PID: 1 Comm: swapper/0 Not tainted 3.10.0-229.4.2.el7.x86_64 #1
[    1.260913]  ffffffff81814288 00000000f2edcb80 ffff88074ed6fd60 ffffffff81604eaa
[    1.260918]  ffff88074ed6fde0 ffffffff815fe71e ffffffff00000010 ffff88074ed6fdf0
[    1.260923]  ffff88074ed6fd90 00000000f2edcb80 00000000f2edcb80 ffff88074ed6fe00
[    1.260928] Call Trace:
[    1.260934]  [<ffffffff81604eaa>] dump_stack+0x19/0x1b
[    1.260938]  [<ffffffff815fe71e>] panic+0xd8/0x1e7
[    1.260943]  [<ffffffff81a4560d>] mount_block_root+0x2a1/0x2b0
[    1.260947]  [<ffffffff81a4566f>] mount_root+0x53/0x56
[    1.260951]  [<ffffffff81a457ae>] prepare_namespace+0x13c/0x174
[    1.260955]  [<ffffffff81a4527b>] kernel_init_freeable+0x203/0x22a
[    1.260959]  [<ffffffff81a449db>] ? initcall_blacklist+0xb0/0xb0
[    1.260963]  [<ffffffff815f33f0>] ? rest_init+0x80/0x80
[    1.260966]  [<ffffffff815f33fe>] kernel_init+0xe/0xf0
[    1.260970]  [<ffffffff81614d3c>] ret_from_fork+0x7c/0xb0
[    1.260973]  [<ffffffff815f33f0>] ? rest_init+0x80/0x80

```

**Comment by jangaraj** [comment]
Thx for report. Could you provide full pastebin.com output from your installation please? Can you try update kernel before installation?

    Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)

Script is not touching root fs mount config, so I guess it's a kernel/OS problem.

**Comment by styk-tv** [comment]
Thank you, your answer actually helped me to find the issue. Since you mentioned kernel might not be updated I thought perhaps after update system needs a reboot. That halted the system as well (before any software was installed). I retraced my steps back to official centos repository >> http://wiki.centos.org/Cloud/AWS (never use anything but those amis) I found that I've been using a different AMI.  Tried CentOS 7 x86_64 (2014_09_29) EBS HVM-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-d2a117ba.2 (ami-e4ff5c93) [this is in eu-west-1] and system updated and rebooted fine. Then installation...... FLAWLESS

You're the man! Nothing wrong with the script.

Cheers,
Peter
